app/controllers/keywords_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class KeywordsController < ApplicationController
+  before_action :ensure_authentication
+  before_action :fetch_file, only: :create
+  before_action :validate_file_type, only: :create
+  before_action :validate_csv, only: :create
+
+  def new
+    render locals: {
+      keyword: Keyword.new
+    }
+  end
+
+  def create
+    Keyword.transaction do
+      CSV.foreach(@file.path, headers: true) do |row|
+        current_user.keywords.create(keyword: row[0])
+      end
```

Comment:
Make sense. But when I tried to use `insert_all`, Rubocop warn that `Avoid using `insert_all` because it skips validations. (convention:Rails/SkipsModelValidations)`

Do you have any idea?

Reply:
I think the issue of `created_at` & `updated_at` could be solved in several ways:
update the table definition to set the default value for those columns as `CURRENT_TIMESTAMP` or modifying the current code as:

```ruby
now = Time.current
bulk_data = []
CSV.foreach(@file.path, headers: true) do |row|
  bulk_data <<  { keyword: row[0], created_at: now, updated_at: now }
end

current_user.keywords.insert_all(bulk_data)
```

one of Rails developer [suggest another way ](https://github.com/rails/rails/pull/35635#issuecomment-478360317)but I don't understand how it works:
```ruby
current_user.keywords.where(created_at: Time.now).insert_all(bulk_data)
```

regarding rubycop, In this case I prefer the performance than the rubocop convention. We can turn off that warning for this part of code 
```ruby
# rubocop:disable Rails/SkipsModelValidations
...code
# rubocop:enable Rails/SkipsModelValidations
```

do you have any inputs @andyduong1920

Reply:
I prefer the bulk insert too but not sure is it easy to perform the sidekiq job per keyword later or not.....

Reply:
@longnd 
In the scraping part I have to create a sidekiq job to process that. As @andyduong1920 prefer me to create 1 job per 1 keyword so I think it will be difficult to perform the sidekiq job if we're using bulk insert.

What do you think?

Reply:
it's up to you to decide @gutakk :D the bulk insert is just a suggestion concering the current implementation for keyword importing.
@andyduong1920 @gutakk  Regarding generating the active job, it depends on how you want to put the data into the job, if you need sth unique that could be easier to archieve using the `create` method, then go for it. Otherwise, if you just need the keyword id (after insert) and the keyword inself, the `insert_all` method supports that. Check this API document https://apidock.com/rails/v6.0.0/ActiveRecord/Persistence/ClassMethods/insert_all
you can pass an additional `returning: %w[ id keyword ]` to the method call to get the id for each keyword

Reply:
Fixed in da1e1ac0e43402e7a11d6dcd0563f882b4cb1433